.github/workflows/test.yml

```diff
@@ -8,27 +8,18 @@ on:
 
 jobs:
   test:
-    runs-on: macos-latest
+    runs-on: macos-12
```

Comment:
- Why locked to 12?
- any reason to keep using macOS when ubuntu is much faster for android emulation nowadays? ([source](https://github.com/ReactiveCircus/android-emulator-runner?tab=readme-ov-file#running-hardware-accelerated-emulators-on-linux-runners))

Reply:
(I understand this was not scope of this PR - just trying to get knowledge so I can make a follow up PR)

Reply:
The last time this pipeline worked it was on a macos that was running java 8 plus it was an Intel Mac. So having this as latest suddenly broke this pipeline.

To actually make this work with the latest (both on the android emulator as well as the mac) there is a need to upgrade DADB to compile on Java 11. But to do this we also need to update the GraalVM native compilation to work with that. TBH I tried this first, but it was not trivial.

Reply:
I see, thanks.

Also, the workflow run is showing as red/failed even though there are no failures? [link](https://github.com/mobile-dev-inc/dadb/actions/runs/10689488332)

Reply:
And why is GraalVM required by this project?

Reply:
> Also, the workflow run is showing as red/failed even though there are no failures? [link (https://github.com/mobile-dev-inc/dadb/actions/runs/10689488332)

I believe because there is a timeout of 15 minutes

> And why is GraalVM required by this project?

Unsure, its unclear to me.

Reply:
FTR, I asked Leland and that's what he said:

> The code itself doesn't depend on GraalVM but we have a test that ensures dadb is compatible with graalvm in case we or other users want to compile it into a native binary using graalvm